exp/controllers/ocimachinepool_controller.go

```diff
@@ -305,13 +304,15 @@ func (r *OCIMachinePoolReconciler) reconcileNormal(ctx context.Context, logger l
 	// Find existing Instance Pool
 	instancePool, err := machinePoolScope.FindInstancePool(ctx)
 	if err != nil {
-		conditions.MarkUnknown(machinePoolScope.OCIMachinePool, infrav2exp.InstancePoolReadyCondition, infrav2exp.InstancePoolNotFoundReason, err.Error())
+		r.Recorder.Event(machinePoolScope.OCIMachinePool, corev1.EventTypeWarning, "ReconcileError", err.Error())
+		conditions.MarkUnknown(machinePoolScope.OCIMachinePool, infrav2exp.InstancePoolReadyCondition, infrav2exp.InstancePoolNotFoundReason, "")
```

Comment:
conditions are stored in the object, so if we change conditions every time there is an error, and if the error has a string which changes every time, this will lead to infinite reconciles. For example, int his case, opc request id changes every time. We had figured this out and fixed it in machine reconciler initially, but since this is such a tricky thing, this was missed in machine pool reconcilers.
Any such errors have to be inserted as events and not in conditions.